compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -589,6 +586,18 @@ public void visit(BLangTupleVariable varNode) {
         typeChecker.checkExpr(varNode.expr, env, varNode.type);
     }
 
+    private BType resolveTupleType(BLangTupleVariable varNode) {
+        List<BType> memberTypes = new ArrayList<>(varNode.memberVariables.size());
+        for (BLangVariable memberVariable : varNode.memberVariables) {
+            if (memberVariable.getKind() == NodeKind.TUPLE_VARIABLE) {
```

Comment:
Need to support record and error binding pattern as well. Please create an issue.